enforcer.go

```diff
@@ -339,18 +339,18 @@ func (e *Enforcer) Enforce(rvals ...interface{}) bool {
 				panic(err)
 			}
 
-			switch result.(type) {
+			switch result := result.(type) {
 			case bool:
-				if !result.(bool) {
+				if !result {
 					policyEffects[i] = effect.Indeterminate
 					continue
 				}
 			case float64:
-				if result.(float64) == 0 {
+				if result == 0 {
 					policyEffects[i] = effect.Indeterminate
 					continue
 				} else {
-					matcherResults[i] = result.(float64)
+					matcherResults[i] = result
 				}
 			default:
 				panic(errors.New("matcher result should be bool, int or float"))
```

Comment:
This error looks misleading as I don't see `case int` in this switch.
It's a separate issue that should be fixed by someone familiar with this code.

Reply:
This is intended, because our library will convert user inputted ``int`` into ``float`` automatically. So our code don't handle ``int`` separately.